internal-api/src/main/java/datadog/trace/bootstrap/instrumentation/api/BlackHoleSpan.java

```diff
@@ -0,0 +1,35 @@
+package datadog.trace.bootstrap.instrumentation.api;
+
+import datadog.trace.api.DDTraceId;
+
+/** An {@link AgentSpan} implementation that stops context propagation. */
+public final class BlackHoleSpan extends NoopSpan {
+  private final DDTraceId traceId;
+
+  public BlackHoleSpan(final DDTraceId traceId) {
+    this.traceId = traceId;
+  }
+
+  @Override
+  public boolean isSameTrace(final AgentSpan otherSpan) {
+    return otherSpan != null
+        && ((traceId != null && traceId.equals(otherSpan.getTraceId()))
+            || otherSpan.getTraceId() == null);
+  }
+
+  @Override
+  public DDTraceId getTraceId() {
+    return traceId;
+  }
+
+  @Override
+  public AgentSpanContext context() {
+    return Context.INSTANCE;
+  }
+
+  public static final class Context extends NoopSpanContext {
```

Comment:
```suggestion
  static final class Context extends NoopSpanContext {
```
AFAICT it should be ok to make this package-private